keps/sig-windows/1981-windows-privileged-container-support/README.md

```diff
@@ -459,87 +463,123 @@ Windows privileged containers will be implemented with [Job Objects](https://doc
+Privileged Windows containers will be unable to align to pod namespaces due to limitiations in the Windows OS. This functionaility may likely be enabled in the future through a new KEP.
+
+Because of this we do not plan on performing **any** valiation around the existing `hostNetowrk` PodSpec field. This will give us the flexability to align privileged containers to host or pod namespaces in the future if/when that functionaility is added. Instead we will rely on documentation to explain 
+
+The PodSpec has a hostNetwork boolean field:
 
 In the case of windows this field would need to be addressed.  During the alpha stage the proposal for this field would be to add documentation that note the field is only applicable to Linux.  In addition the check for hostnetwork mode will need to be updated to check the privileged flag on windows PodSpec.  This will be implemented in the following way, with validation loosened in the future if required:
- - this pod must run on a windows host, and kubelets must reject it if not on windows hosts
- - all pods marked privileged on windows must have host network enabled, if not the pod does not validate
 
-In beta there is a possibility to enable the privileged container to be a part of a different network component.  If this feature is enabled we will use the existing Pod HostNetwork field to enable/disable.
+- this pod must run on a windows host, and kubelets must reject it if not on windows hosts
+- all pods marked privileged on windows must have host network enabled, if not the pod does not validate
+
+### Kubelet Implementation Details
 
-#### <b> ContainerD Support Only</b>
-There are no plans to update Docker to have support for Privileged containers due to requirements for support HCSv2.  The additional fields for docker would be ignored. The default value should be false.
+Kubelet will pass privileged flag from `WindowsSecurityCOntextOptions` to the appropraite CRI layer calls.
 
-Need to investigate if the choice of base image would cause errors when pulling with docker.
+*Note:* For alpha kubelet may add well-known annotaitons to CRI calls if privileged flags are set.
 
-#### <b> Feature Gates</b>
-Although there are no changes required to the kubernetes API, the ability to pass the the privileged flag to the CRI should be behind a feature gate in Kubelet:  
+Add functionality to Kuberuntime_sandbox to:
 
-https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/#feature-stages
+- Split out the linux sandbox creation and add [windows sandbox creation](https://github.com/kubernetes/kubernetes/blob/a9f1d72e1de6450b890a0c0e451725468f54f515/pkg/kubelet/kuberuntime/kuberuntime_sandbox.go#L136)
+- Configure all privileged Windows pods to join the [host network](https://github.com/kubernetes/kubernetes/blob/a9f1d72e1de6450b890a0c0e451725468f54f515/pkg/kubelet/kuberuntime/kuberuntime_sandbox.go#L98)
 
+#### ContainerD Support Only
```

Comment:
should this be "CRI Support Only"?

Reply:
Yes, I will update (I was thinking of making this update anyway).
Most of the platform changes are in hcsshim which is invoked by containerd. Right now contianerd is the only CRI implementation for Windows but there would be nothing stopping future container runtimes from supporting this.